pyproject.toml

```diff
@@ -0,0 +1,27 @@
+[build-system]
+requires = ["scikit-build-core >=0.4.3", "nanobind >=1.3.1"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "py-osrm"
+version = "0.0.1"
+description = "Python bindings for the osrm-backend project"
+readme = "README.md"
+requires-python = ">=3.7"
+classifiers = [
+    "License :: BSD",
+]
+
+[project.urls]
+repository = "https://github.com/gis-ops/py-osrm"
+"osrm-backend repository" = "https://github.com/Project-OSRM/osrm-backend"
+
+[tool.scikit-build]
+# Protect the configuration against future changes in scikit-build-core
+minimum-version = "0.4"
+
+# Setuptools-style build caching in a local directory
+build-dir = "build/{wheel_tag}"
+
+# Build stable ABI wheels for CPython 3.12+
+wheel.py-api = "cp312"
```

Comment:
I didn't quite understand the purpose of this by the docs: https://github.com/scikit-build/scikit-build-core#configuration.

So this would be ABI compatible with future versions Python? If we put 3.7 here, it'd be compatible with 3.7+? I always figured Python is ABI incompatible between minor versions, that's why we always need to build for each one separately (via e.g. cibuildwheel)